packages/sw-background-sync-queue/src/lib/request-manager.js

```diff
@@ -0,0 +1,76 @@
+import {putResponse} from './response-manager';
+import {getFetchableRequest} from './queue-utils';
+import {tagNamePrefix} from './constants';
+/**
+ * Class to handle all the request related
+ * transformations, replaying, event handling
+ * broadcasting back to controlled pages etc.
+ * @class
+ */
+class RequestManager {
+	/**
+	 * Initializes the request manager
+	 * stores the callbacks object, maintins config and
+	 * attaches event handler
+	 * @param {any} {callbacks, queue}
+	 *
+	 * @memberOf RequestManager
+	 */
+	constructor({callbacks, queue}) {
```

Comment:
Consider using the [`assert` library](https://github.com/GoogleChrome/sw-helpers/blob/master/lib/assert.js) to validate these parameters if they're required.